src/main/java/org/mamute/dao/QuestionDAO.java

```diff
@@ -152,17 +153,15 @@ public Question load(Question question) {
 				.list();
 	}
 	
-	public List<Question> top(String section, int count, DateTime since) {
+	public List<Question> top(Tab tab, int count, DateTime since) {
 		Order order;
-		if (section.equals("viewed")) {
-			order = Order.desc("q.views");
-		}
-		else if (section.equals("answered")) {
-			order = Order.desc("q.answerCount");
-		}
-		else /*if (section.equals("voted"))*/ {
-			order = Order.desc("q.voteCount");
+
+		switch (tab.getType()) {
+			case VIEWED:    order = Order.desc("q.views");          break;
+			case ANSWERED:  order = Order.desc("q.answerCount");    break;
+			default:        order = Order.desc("q.voteCount");      break;
 		}
```

Comment:
I see your point, however don't you think that would be too much of a mixing concerns? Why would a Tab object need to know about SQL queries? What about rather changing the getType() to getOrder() which would return a order enum, getting rid of type all together? This still gives better idea about what tab is doing, but does not make a dependency between Tag and the SQL